test-specifications/token-service/tokenCreateTransaction.md

```diff
@@ -0,0 +1,1158 @@
+| 3       | Creates a token with a valid ED25519 private key as its fee schedule key                                                | name="testname", symbol="testsymbol", treasuryAccountId=<OPERATOR_ACCOUNT_ID>, feeScheduleKey=<VALID_ED25519_PRIVATE_KEY>         | The token creation succeeds and the token has the corresponding new ED25519 public key as its fee schedule key.        | N                 |
+| 4       | Creates a token with a valid ECDSAsecp256k1 private key as its fee schedule key                                         | name="testname", symbol="testsymbol", treasuryAccountId=<OPERATOR_ACCOUNT_ID>, feeScheduleKey=<VALID_ECDSA_SECP256K1_PRIVATE_KEY> | The token creation succeeds and the token has the corresponding new ECDSAsecp256k1 public key as its fee schedule key. | N                 |
+| 5       | Creates a token with a valid KeyList of ED25519 and ECDSAsecp256k1 private and public keys as its fee schedule key      | name="testname", symbol="testsymbol", treasuryAccountId=<OPERATOR_ACCOUNT_ID>, feeScheduleKey=<VALID_KEYLIST>                     | The token creation succeeds and the token has the new KeyList as its fee schedule key.                                 | N                 |
+| 6       | Creates a token with a valid KeyList of nested Keylists (three levels) as its fee schedule key                          | name="testname", symbol="testsymbol", treasuryAccountId=<OPERATOR_ACCOUNT_ID>, feeScheduleKey=<VALID_NESTED_KEYLIST>              | The token creation succeeds and the token has the new nested KeyList as its fee schedule key.                          | N                 |
+| 7       | Creates a token with a valid ThresholdKey of ED25519 and ECDSAsecp256k1 private and public keys as its fee schedule key | name="testname", symbol="testsymbol", treasuryAccountId=<OPERATOR_ACCOUNT_ID>, feeScheduleKey=<VALID_THRESHOLD_KEY>               | The token creation succeeds and the token has the new ThresholdKey as its fee schedule key.                            | N                 |
+| 8       | Creates a token with an invalid key as its fee schedule key                                                             | name="testname", symbol="testsymbol", treasuryAccountId=<OPERATOR_ACCOUNT_ID>, feeScheduleKey=<INVALID_KEY>                       | The token creation fails with an SDK internal error.                                                                   | N                 |
+
+#### JSON Request Example
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 12,
+  "method": "createToken",
+  "params": {
+    "name": "testname",
+    "symbol": "testsymbol",
+    "treasuryAccountId": "0.0.2",
+    "feeScheduleKey": "302a300506032b6570032100e9a0f9c81b3a2bb81a4af5fe05657aa849a3b9b0705da1fb52f331f42cf4b496"
+  }
+}
+```
+
+#### JSON Response Example
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 12,
+  "result": {
+    "tokenId": "0.0.541",
+    "status": "SUCCESS"
+  }
+}
+```
+
+### **Custom Fees:**
```

Comment:
Can add test case for response code `CUSTOM_FEE_NOT_FULLY_SPECIFIED` where a custom fee schedule entry did not specify either a fixed or fractional fee

Reply:
I don't think it's possible to get this response code using the SDKs... I double checked with services, this response code is given when a `CustomFee` protobuf does not have its fee `oneof` set. Since its impossible in the SDKs to instantiate a `CustomFee` that isn't a `CustomFixedFee`, `CustomFractionalFee`, or `CustomRoyaltyFee`, then I think its impossible to get that response code using the SDKs.

Reply:
Oh I see.